### PR TITLE
Feature/file format doc

### PR DESCRIPTION
**Pull request by blairlyons**
this PR adds documentation of the .simularium JSON format. I put it in a markdown file at the root of the repo, open to suggestions if there's a better place to put it.

### CODE REVIEW

**Comment by toloudis** [comment]
We have it documented in typescript also, in the viewer.  That is a definitive "if this is broken nothing will work" version of the file format.  Because of that, I would compare what you have here with that one.

I wonder if we can auto-gen some of this from the viewer's type definitions, or produce a more formal type definition here.. like one of these https://json-schema.org/

**Review by toloudis — APPROVED** [review]

File: file_format.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+# .simularium JSON Format
+
+JSON files accepted by the simularium-viewer contain the following data in JSON format:
+* **trajectory info**
+  * timeStepSize - the amount of time that passes in the simulation for each time step
```
Comment (toloudis) [inline file_format.md:5]: version is missing.  This doc is documenting version 1 and should make that clear.

Comment (blairlyons) [reply file_format.md:5]: whoops

**Comment by blairlyons** [comment]
> We have it documented in typescript also, in the viewer. That is a definitive "if this is broken nothing will work" version of the file format. Because of that, I would compare what you have here with that one.

@toloudis VisDataMessage and CachedObservables don't have "version" in the viewer, is that ok?
https://github.com/allen-cell-animated/simularium-viewer/blob/f5b870b0d597a23a9708dcb5aa7ad72bbce19cfd/src/simularium/types.ts#L7

**Comment by toloudis** [comment]
also on the subject of generating docs for data types:  http://typedoc.org/

**Comment by toloudis** [comment]
> > We have it documented in typescript also, in the viewer. That is a definitive "if this is broken nothing will work" version of the file format. Because of that, I would compare what you have here with that one.
> 
> @toloudis VisDataMessage and CachedObservables don't have "version" in the viewer, is that ok?
> https://github.com/allen-cell-animated/simularium-viewer/blob/f5b870b0d597a23a9708dcb5aa7ad72bbce19cfd/src/simularium/types.ts#L7

I'm not sure if that is ok!   The interfaces in the viewer code that use it might not yet be using the version field. We should put it in there, tho, if it's really there.

File: file_format.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                "pdb" : "agent1.pdb",  // optional
+                "mesh" : "agent1.obj"  // optional
+            },
+            "1" : {
+                "name" : "agent1#bound",
+                "pdb" : "agent1.pdb",  // optional
+                "mesh" : "agent1.obj"  // optional
+            },
+            "2" : {
+                "name" : "agent2",     // optional
```
Comment (EricJIsaac) [inline file_format.md:83]: Is this optional?

**Review by EricJIsaac — CHANGES_REQUESTED** [review]

File: file_format.md (hunk 2 above)
Comment (blairlyons) [reply file_format.md]: it does fallback to ID, but I didn't mean to add optional to this one

**Review by EricJIsaac — APPROVED** [review]

**Pull request merged**